### ISSUE TITLE
Your Google authorisation token needs updating.

### ISSUE THREAD

**Issue by markcs** [phw198/OutlookGoogleCalendarSync#299]
I'm trying to set up OutlookGoogleCalendarSync, but when I try to connect the application to my gmail calendar, I keep getting the following message after pasting in the token I am given by Google:

2017-08-18 13:47:29,393 INFO  1 OutlookGoogleCalendarSync.MainForm [0] -  Your Google authorisation token needs updating.
The process to reauthorise access to your Google account will now begin...

I can never fully authorise my account and so never get a list of google calendars.

**Comment by phw198** [comment]
Hi @markcs 

:spiral_notepad: Please could you provide your OGcalsync.log file [[see instructions](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#posting-your-ogcalsynclog-file)]

**Comment by markcs** [comment]
@phw198 I sent you an email with the Gist URL

**Comment by DougGreening** [comment]
First time I ran this issue, I checked for updates and updated appropriately.   After update it ran fine for a few hours and now running into the same issue again.   Checking updates right now failed one, something about zip consistency.
It appears that application id may be changing with each update?

BTW, this application has been great and exactly what I have been looking for.  I sincerely hope you can address this issue.

**Comment by phw198** [comment]
:package: Could you try hotfix [v2.6.0.1.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/1235815/v2.6.0.1.zip) [[see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)]. 

This should open a browser window with a URL that starts with `https://www.googleapis.com/userinfo/email?alt=xml&access_token=` - does the page display your Google email id, or an error (eg insufficient permission)? If you have other browsers installed, do you get the same result in all browsers?

**Comment by markcs** [comment]
Now it firstly generates a token at accounts.google.com, which is copied/pasted into the application.  Then I get a message saying the token has expired, and a new page opens at the googleapis url you listed.

But I get this error now:
This site can’t be reached

The webpage at https://www.googleapis.com/userinfo/email?alt=xml&access_token=xx might be temporarily down or it may have moved permanently to a new web address.
ERR_INVALID_RESPONSE

**Comment by phw198** [comment]
OK, that's strange. You should now also have an extra line in your log file between these two lines:
```
Refresh and Access token successfully retrieved.
Google API keys and refresh token are being updated.
```
Could you let me know what that says?

Can you also try changing the URL in the web browser when it pops up to https://www.googleapis.com/userinfo/email?access_token=xx (remove the `alt=xml&` part).

**Comment by phw198** [comment]
Turns out the `userinfo` API is deprecated and people have reported differing success continuing to use it. Alpha release v2.6.1 is now using the new methodology, so could you upgrade to that and see if it resolves the problem?

@DougGreening Sounds like you're reporting an issue with checking/applying an upgrade? If so, that should be raised under a new issue...

**Comment by markcs** [comment]
I'm unable to authenticate to Google....
```
2017-08-24 11:30:06,189 INFO  6 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  No Google credentials file available - need user authorisation for OGCS to manage their calendar.
2017-08-24 11:30:06,212 ERROR  6 OutlookGoogleCalendarSync.OGCSexception [0] -  System.IO.FileLoadException: Could not load file or assembly 'System.Net.Http.Primitives, Version=1.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
2017-08-24 11:30:06,212 ERROR  6 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131040;-2146234304
2017-08-24 11:30:06,228 ERROR  1 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  Problem encountered instantiating Authenticator.
2017-08-24 11:30:06,230 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.NullReferenceException: Object reference not set to an instance of an object.
2017-08-24 11:30:06,231 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80004003;-2147467261
2017-08-24 11:30:06,237 INFO  8 OutlookGoogleCalendarSync.MainForm [0] -  Unable to authenticate with Google. The following error occurred:
2017-08-24 11:30:06,237 INFO  15 OutlookGoogleCalendarSync.MainForm [0] -  Could not load file or assembly 'System.Net.Http.Primitives, Version=1.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

**Comment by phw198** [comment]
@markcs Sorry - this is a problem with the packaging of v2.6.1 that's being look at under issue #305.

**Comment by markcs** [comment]
@phw198 I updated the config file, but now get this issue:
```
2017-08-24 18:25:53,048 INFO  5 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  No Google credentials file available - need user authorisation for OGCS to manage their calendar.
2017-08-24 18:26:17,017 INFO  16 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  Refresh and Access token successfully retrieved.
2017-08-24 18:26:17,027 DEBUG  16 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  Access token expires 24/08/2017 19:26:16
2017-08-24 18:26:18,330 INFO  16 OutlookGoogleCalendarSync.GoogleOgcs.ApiKeyring [0] -  Google API keys and refresh token are being updated.
2017-08-24 18:26:21,776 ERROR  1 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  Problem encountered instantiating Authenticator.
2017-08-24 18:26:21,777 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.NullReferenceException: Object reference not set to an instance of an object.
2017-08-24 18:26:21,777 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80004003;-2147467261
```

**Comment by phw198** [comment]
I think your IP or domain or something is being blocked by Google. We know the code works now as reported in issue #305 and the API keys embedded in OGCS are being used OK by everyone else. 

Some (repeated) steps to troubleshoot:
1. Go to `%APPDATA%\Outlook Google Calendar Sync` and delete `settings.xml` and `Google.Apis.Auth.OAuth2.Responses.TokenResponse-user`
1. Open OGCS and try to retrieve Google calendars
    1. Go through authorisation steps - I expect it to fail
1. Open the new `Google.Apis.Auth.OAuth2.Responses.TokenResponse-user` file and copy out of it the value for `access_token`
1. Put that value on the end of https://www.googleapis.com/plus/v1/people/me?fields=emails&access_token= and paste it into your web browser.
1. Do you get the 403 error? Are there any more details?

Another thing you could try is to set up your own Google API keys and try using those - click the advanced checkbox on the Google settings tab in OGCS and follow the link in to the Google API Console. You'll need to enable the Calendar API and Google+

**Comment by markcs** [comment]
I've got it working and have successfully authenticated now.  Thanks!

**Comment by phw198** [comment]
Great! What was the solution? Interested to know in case others have the same problem.

**Comment by phw198** [comment]
@markcs Any info you can provide?

**Comment by markcs** [comment]
Not really unfortunately.  I don't really know why the authorisation started working.  Sorry, not much use I know!
